pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -240,8 +213,17 @@ func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Wo
 		attempt := int32(1)
 		shouldCreatePr := false
 		if exists {
-			attempt = getAttempt(ctx, oldPr, wl.Name, checkName)
-			if apimeta.IsStatusConditionTrue(oldPr.Status.Conditions, autoscaling.Failed) {
+			switch {
+			case isCapacityRevoked(oldPr):
+				if workload.IsActive(wl) && !workload.IsFinished(wl) {
+					workload.SetEvictedCondition(wl, kueue.WorkloadEvictedByDeactivation, fmt.Sprintf("Deactivating workload because capacity for %v has been revoked", oldPr.Name))
```

Comment:
you need an API call for updating the Workload status, separate from updating the spec below.

And this is a sign that we are missing an check in the integration test.

Reply:
Fixed